### ISSUE TITLE
🐛 tsconfig.json should be ignored, but it now reports problems

### ISSUE THREAD

**Issue by zhoujian-official** [biomejs/biome#933]
### Environment information

```block
CLI:
  Version:                      1.4.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.3"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

1. open tsconfig.json in vscode. this file should be [ignored by biome](https://biomejs.dev/guides/how-biome-works/#known-files) and throw no error.
2. 
![image](https://github.com/biomejs/biome/assets/35226412/dc95898c-77d2-46e7-bb0f-64974b465d96)

### Expected result

do not show lint error of tsconfig.json

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Do you see these errors even when you run the CLI?

**Comment by umstek** [comment]
Back in 1.9.4

$ biome check --write .
./tsconfig.node.json:9:3 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ JSON standard does not allow comments.
  
     7 │                "skipLibCheck": true,
     8 │ 
   > 9 │                /* Bundler mode */
       │                ^^^^^^^^^^^^^^^^^^
    10 │                "moduleResolution": "bundler",
    11 │                "allowImportingTsExtensions": true,
  

./tsconfig.node.json:10:3 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ End of file expected

**Comment by stychu** [comment]
1.9.4 same here but the weird thing is that I scaffolded tanstack router example and it doesnt complain but when I make a monorepo with pnpm and having biome at root it start to yell about comments in tsconfig files...

**Comment by sep2** [comment]
This should also be ignored by `tsconfig.app.json` & `tsconfig.node.json` and etc.
